crates/store/re_sorbet/src/sorbet_batch.rs

```diff
@@ -173,20 +182,27 @@ impl SorbetBatch {
 
         let sorbet_schema = SorbetSchema::try_from(batch.schema_ref().as_ref())?;
 
-        for (field, column) in itertools::izip!(
+        let new_fields = itertools::izip!(
+            batch.schema_ref().fields(),
             sorbet_schema.columns.arrow_fields(batch_type),
             batch.columns()
-        ) {
-            debug_assert_eq!(field.data_type(), column.data_type());
-        }
+        )
+        .map(|(old_field, mut new_field, column)| {
+            debug_assert_eq!(new_field.data_type(), column.data_type());
+
+            let mut metadata = old_field.metadata().clone();
+            metadata.extend(new_field.metadata().clone()); // overwrite old with new
```

Comment:
Would that work in a setting where we change keys as part of our migration? Because that currently happens on `main`.